syntax = "proto3";

package mavsdk.rpc.geofence;

option java_package = "io.mavlink.mavsdk.geofence";
option java_outer_classname = "GeofenceProto";

service GeofenceService {
    rpc UploadGeofence(UploadGeofenceRequest) returns(UploadGeofenceResponse) {}
}


message Point {
    double latitude_deg = 1;
    double longitude_deg = 2;
}

message Polygon {
    enum Type {
        INCLUSION = 0;
        EXCLUSION = 1;
    }

    repeated Point points = 1;
    Type type = 2;
}
    
message UploadGeofenceRequest {
    repeated Polygon polygons = 1;
}

message UploadGeofenceResponse {
    GeofenceResult geofence_result = 1;
}

message GeofenceResult {
    Result result = 1;
    string result_str = 2;

    enum Result {
        UNKNOWN = 0;
        SUCCESS = 1;
        ERROR = 2;
        TOO_MANY_GEOFENCE_ITEMS = 3;
        BUSY = 4;
        TIMEOUT = 5;
        INVALID_ARGUMENT = 6;
    }
}
